### DISCUSSION TITLE
resource check interval managed by "resource.check_every" somehow slip

### DISCUSSION THREAD

**Discussion by syurui1** [concourse/concourse#5763] [General]
【environment】
version: 6.3(latest so far)
【question】
I wonder how I can manage the interval on which to check for new versions of the resource precisely.
It seems that there is a time lag between the resource check interval managed by "resource.check_every"  and actual behavior.
For example, the following table describes degree of lag in my environment.
I monitor the actual interval by gazing "checked Ns ago" on the resource GUI.
| resource.check_every| actual check frequency |
| ------------- | ------------- |
| default  | 1m8s  |
| 1m20s   | 1m34s  |
| 1m10s   | 1m20s  |
| 1m   | 1m8s  |
| 50s   | 1m8s  |
| 40s   | 1m3s  |
| 30s   | 43s  |
| 20s   | 30s  |
| 10s   | 24s  |
| 1s   | 14s  |
(figures on actual check frequency are slightly different every time. It is OK.)

As you can see from this above table, there are lags between my configuration and actual behavior. In addition, I can't find the regularity of spacing. Why there are a big difference between "40s" and "30s"?

Is that expected behavior on the current version? And is there any way to manage them more precisely?
Now I'm trying to figure out the reason for this behavior by reading code base, but I would appreciate if someone helps me.

If you need further information, please feel free to ask me.

**Comment by vito** [comment]
This is expected behavior. Checking is performed in batches every 10 seconds or so by default. `check_every` is not intended to be a precise timer, it's meant to be a threshold after which a check will be performed. It's normally used to set higher values, not lower, so it tends to not be a big issue.

In addition to the 10 seconds delay, if there are equivalent `time` resource definitions across pipelines (say `interval: 10m`) they will currently wait for each other to stop using the same `check` container, which can mean it sometimes takes (N * 10 seconds) for the check to happen. #5730 explains this, though #3079 may be a more long-term solution.
